azurerm/resource_arm_action_group.go

```diff
@@ -0,0 +1,313 @@
+	}
+
+	if err = d.Set("sms_receiver", flattenActionGroupSmsReceiver(resp.SmsReceivers)); err != nil {
+		return fmt.Errorf("Error setting `sms_receiver` of action group %s (resource group %s): %+v", name, resGroup, err)
+	}
+
+	if err = d.Set("webhook_receiver", flattenActionGroupWebHookReceiver(resp.WebhookReceivers)); err != nil {
+		return fmt.Errorf("Error setting `webhook_receiver` of action group %s (resource group %s): %+v", name, resGroup, err)
+	}
+
+	flattenAndSetTags(d, resp.Tags)
+
+	return nil
+}
+
+func resourceArmActionGroupDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).actionGroupsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return fmt.Errorf("Error parsing action group resource ID \"%s\" during delete: %+v", d.Id(), err)
+	}
+	resGroup := id.ResourceGroup
+	name := id.Path["actionGroups"]
+
+	resp, err := client.Delete(ctx, resGroup, name)
+	if err != nil {
+		if response.WasNotFound(resp.Response) {
+			return nil
+		}
+		return fmt.Errorf("Error deleting action group %s (resource group %s): %+v", name, resGroup, err)
+	}
+
+	return nil
+}
+
+func expandActionGroupEmailReceiver(v []interface{}) *[]insights.EmailReceiver {
+	receivers := make([]insights.EmailReceiver, 0)
+	for _, receiverValue := range v {
+		val := receiverValue.(map[string]interface{})
+		receiver := insights.EmailReceiver{
+			Name:         utils.String(val["name"].(string)),
+			EmailAddress: utils.String(val["email_address"].(string)),
+		}
+		receivers = append(receivers, receiver)
+	}
+	return &receivers
+}
+
+func expandActionGroupSmsReceiver(v []interface{}) *[]insights.SmsReceiver {
+	receivers := make([]insights.SmsReceiver, 0)
+	for _, receiverValue := range v {
+		val := receiverValue.(map[string]interface{})
+		receiver := insights.SmsReceiver{
+			Name:        utils.String(val["name"].(string)),
+			CountryCode: utils.String(val["country_code"].(string)),
+			PhoneNumber: utils.String(val["phone_number"].(string)),
+		}
+		receivers = append(receivers, receiver)
+	}
+	return &receivers
+}
+
+func expandActionGroupWebHookReceiver(v []interface{}) *[]insights.WebhookReceiver {
+	receivers := make([]insights.WebhookReceiver, 0)
+	for _, receiverValue := range v {
+		val := receiverValue.(map[string]interface{})
+		receiver := insights.WebhookReceiver{
+			Name:       utils.String(val["name"].(string)),
+			ServiceURI: utils.String(val["service_uri"].(string)),
+		}
+		receivers = append(receivers, receiver)
+	}
+	return &receivers
+}
+
+func flattenActionGroupEmailReceiver(receivers *[]insights.EmailReceiver) []interface{} {
+	result := make([]interface{}, 0)
+	if receivers != nil {
+		for _, receiver := range *receivers {
+			val := make(map[string]interface{}, 0)
+			val["name"] = *receiver.Name
+			val["email_address"] = *receiver.EmailAddress
```

Comment:
we should nil-check around both of these properties as there's potential crashes here